spec/api/definitions_spec.cr

```diff
@@ -392,6 +392,58 @@ describe LavinMQ::HTTP::Server do
     ensure
       s.vhosts["/"].delete_policy("export_p2")
     end
+
+    describe "user tags and vhost access" do
+      it "export vhost definitions as management user" do
+        s.users.delete("guest")
+        s.users.create("other_name", "guest", [LavinMQ::Tag::Management], save: false) # Will be the new default_user
+        s.vhosts.create("new")
+        s.users.add_permission("other_name", "new", /.*/, /.*/, /.*/)
+        headers = HTTP::Headers{"Authorization" => "Basic b3RoZXJfbmFtZTpndWVzdA=="}
+        response = get("/api/definitions/new", headers: headers)
+        response.status_code.should eq 200
+      ensure
+        s.users.delete("other_name", save: false)
+        s.vhosts.delete("new")
+        s.users.create("guest", "guest", [LavinMQ::Tag::Administrator])
+        s.vhosts.each_key { |name| s.users.add_permission("guest", name, /.*/, /.*/, /.*/) }
+      end
+
+      it "should refuse vhost access" do
+        s.users.delete("guest")
+        s.users.create("other_name", "guest", [LavinMQ::Tag::Management], save: false) # Will be the new default_user
+        s.vhosts.create("new")
+        headers = HTTP::Headers{"Authorization" => "Basic b3RoZXJfbmFtZTpndWVzdA=="}
+        response = get("/api/definitions/new", headers: headers)
+        response.status_code.should eq 401
+        body = JSON.parse(response.body)
+        body["reason"].should eq "Access refused"
+      ensure
+        s.users.delete("other_name", save: false)
+        s.vhosts.delete("new")
+        s.users.create("guest", "guest", [LavinMQ::Tag::Administrator])
+        s.vhosts.each_key { |name| s.users.add_permission("guest", name, /.*/, /.*/, /.*/) }
+      end
+
+      it "should refuse user tag access" do
+        s.users.delete("guest")
+        # Http tag override refuse_unless_management
+        s.users.create("other_name", "guest", [LavinMQ::Tag::Http], save: false) # Will be the new default_user
+        s.vhosts.create("new")
+        s.users.add_permission("other_name", "new", /.*/, /.*/, /.*/)
+        headers = HTTP::Headers{"Authorization" => "Basic b3RoZXJfbmFtZTpndWVzdA=="}
+        response = get("/api/definitions/new", headers: headers)
```

Comment:
New user without tags, have no access to API at all, hence issue with login refused.

Setting tag to Http (to be able to make the API call), will override management and allow exporting vhost definitions. https://github.com/cloudamqp/lavinmq/blob/main/src/lavinmq/http/controller.cr#L194